railties/lib/rails/application/bootstrap.rb

```diff
@@ -104,6 +105,20 @@ module Bootstrap
       initializer :set_secrets_root, group: :all do
         Rails::Secrets.root = root
       end
+
+      # Sets the Rails.application.url config
+      initializer :initialize_url, group: :all, after: :one do |app|
+        if app.config.application_url
+          url = ActionDispatch::Http::URI.new(app.config.application_url)
+          Rails.application.url = url
+          if app.default_url_options.blank?
+            app.default_url_options = { host: url.host, protocol: url.scheme, port: url.port }
```

Comment:
Since it's the same map for both conditions, how about pull it out of the branch?